From 17f5d2f574236f8c3459f9efadef2f0f6220a4dd Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 13 Aug 2018 15:46:53 -0700
Subject: [PATCH] Check for libegl using pkg-config

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
[Roman: patch has been rebased to 223817ee3798 ("Add Wayland support")
 trivial merge conflicts resolved]
Signed-off-by: Roman Stratiienko <roman.stratiienko@globallogic.com>
---
 wscript | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/wscript b/wscript
index 0f6ec53..401f62e 100644
--- a/wscript
+++ b/wscript
@@ -174,14 +174,9 @@ def configure_raspberrypi_device(conf, platform):
 	else:
 		conf.fatal('Unsupported Raspberry Pi platform "%s"' % platform)
 		return
-	conf.check_cxx(mandatory = 1, lib = ['GLESv2', 'EGL', 'bcm_host'], uselib_store = 'EGL')
+	conf.check_cfg(package='egl', args='--libs --cflags')
 	import os
 	sysroot = conf.options.sysroot + conf.options.prefix
-        std_include_path = os.path.join(sysroot, 'include')
-	vcos_pthread_path = os.path.join(sysroot, 'include/interface/vcos/pthreads')
-	vcms_host_path = os.path.join(sysroot, 'include/interface/vmcs_host/linux')
-	if not conf.check_cxx(mandatory = 0, header_name = ['vcos_platform_types.h', 'EGL/egl.h', 'bcm_host.h'], includes = [vcos_pthread_path, vcms_host_path, std_include_path], uselib_store = 'EGL'):
-		conf.fatal('Check if --prefix and --sysroot are set correctly.')
 	conf.env['WITH_APIS'] = []
 	if check_gles2(conf):
 		conf.env['WITH_APIS'] += ['GLES1', 'GLES2']
-- 
2.19.1
